Index: zeitgeist-0.9.14/data/completions/zeitgeist-daemon
===================================================================
--- zeitgeist-0.9.14.orig/data/completions/zeitgeist-daemon	2013-06-17 08:40:08.000000000 +0200
+++ zeitgeist-0.9.14/data/completions/zeitgeist-daemon	2013-10-25 23:16:30.874139250 +0200
@@ -5,10 +5,10 @@
 #
 # By Siegfried-Angel Gevatter Pujals <siegfried@gevatter.com>
 
-have zeitgeist-daemon &&
+_have zeitgeist-daemon &&
 _zeitgeist_daemon()
 {
 	local cur=${COMP_WORDS[COMP_CWORD]}
 	COMPREPLY=($(compgen -W "`zeitgeist-daemon --shell-completion`" -- $cur))
-}
-[ "${have:-}" ] && complete -F _zeitgeist_daemon -o default zeitgeist-daemon
+} &&
+complete -F _zeitgeist_daemon -o default zeitgeist-daemon
